pkg/controller/storageprofile-controller.go

```diff
@@ -279,52 +293,83 @@ func isNoProvisioner(name string, cl client.Client) bool {
 	return storageClass.Provisioner == "kubernetes.io/no-provisioner"
 }
 
-func (r *StorageProfileReconciler) computeMetrics() error {
-	if err := r.checkIncompleteProfiles(); err != nil {
-		return err
-	}
-	if err := r.checkDefaultVirtStorageClasses(); err != nil {
-		return err
+func (r *StorageProfileReconciler) computeMetrics(profile *cdiv1.StorageProfile, sc *storagev1.StorageClass) error {
+	if profile.Status.StorageClass == nil || profile.Status.Provisioner == nil {
+		return nil
 	}
 
-	return nil
-}
+	storageClass := *profile.Status.StorageClass
+	provisioner := *profile.Status.Provisioner
 
-func (r *StorageProfileReconciler) checkDefaultVirtStorageClasses() error {
-	defaultVirtStorageClassCount := 0
-	storageClassList := &storagev1.StorageClassList{}
-	if err := r.client.List(context.TODO(), storageClassList); err != nil {
+	// We don't count explicitly unsupported provisioners as incomplete
+	_, found := storagecapabilities.UnsupportedProvisioners[*profile.Status.Provisioner]
+	isComplete := found || !isIncomplete(profile.Status.ClaimPropertySets)
+	isDefault := sc.Annotations[cc.AnnDefaultStorageClass] == "true"
+	isVirtDefault := sc.Annotations[cc.AnnDefaultVirtStorageClass] == "true"
+	isRWX := hasRWX(profile.Status.ClaimPropertySets)
+	isSmartClone, err := r.hasSmartClone(profile)
+	if err != nil {
 		return err
 	}
-	for _, sc := range storageClassList.Items {
-		if sc.Annotations[cc.AnnDefaultVirtStorageClass] == "true" {
-			defaultVirtStorageClassCount++
+
+	// Setting the labeled Gauge to 1 will not set the older Gauges to 0, so we need to explicitly set them
+	scLabels := createLabels(storageClass, provisioner, isComplete, isDefault, isVirtDefault, isRWX, isSmartClone)
+
+	// FIXME: do it smarter
+	for i := 0; i < 1<<5; i++ {
```

Comment:
sure you want to go there? tbh, I don't fully understand the problem domain but would expect a Prometheus helper around this
@machadovilaca Hey! do you have a tip here maybe?

Reply:
will .Reset() and then .Set() not work here?

Reply:
`MetricVec.Reset()` will delete **all** metrics in the vector, so even if all labels are the same before and after reconcile, **all** the storage profiles metrics will be deleted in each reconcile of a specific storage profile, and only this specific storage profiles metric will be `Set()`.

Reply:
Ok, got rid of the silly loop. Used `DeletePartialMatch()` to delete any possible previous metric of the specific storage profile, before setting the updated metric.